eclass/debian.eclass

```diff
@@ -0,0 +1,85 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: debian.eclass
+# @MAINTAINER:
+# Sam James <sam@gentoo.org>
+# @SUPPORTED_EAPIS: 8
+# @BLURB: Convenience eclass for easily using Debian patchsets as upstream
+# @DESCRIPTION:
+# TODO
+
+# @ECLASS_VARIABLE: DEBIAN_PN
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Debian name for the package. Defaults to ${PN}.
+: ${DEBIAN_PN:=${PN}}
+
+# @ECLASS_VARIABLE: DEBIAN_PV
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Debian patch level. Defaults to ${PV/*_p}.
+: ${DEBIAN_PV:=${PV/*_p}}
+
+# @ECLASS_VARIABLE: DEBIAN_ORIG
+# @DEFAULT_UNSET
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Controls whether the "original" upstream sources (unpatched) are fetched
+# from Debian or not. Defaults to no.
+
+# @ECLASS_VARIABLE: DEBIAN_ORIG_PV
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Controls the PV used without the patchlevel. Defaults to ${PV/_p*}.
+: ${DEBIAN_ORIG_PV:=${PV/_p*}}
+
+# @ECLASS_VARIABLE: DEBIAN_ORIG_EXT
+# @DEFAULT_UNSET
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Controls the Debian "original" upstream sources (unpatched) file extension.
+# Defaults to gz.
+: ${DEBIAN_ORIG_EXT:=gz}
+
+if [[ -n ${DEBIAN_ORIG} ]] ; then
+	SRC_URI+=" mirror://debian/pool/main/${DEBIAN_PN::1}/${DEBIAN_PN}/${DEBIAN_PN}_${DEBIAN_ORIG_PV}.orig.tar.${DEBIAN_ORIG_EXT}"
+	S="${WORKDIR}"/${DEBIAN_PN}-${DEBIAN_ORIG_PV}
+fi
+
+SRC_URI+=" mirror://debian/pool/main/${DEBIAN_PN::1}/${DEBIAN_PN}/${DEBIAN_PN}_${DEBIAN_ORIG_PV}-${DEBIAN_PV}.debian.tar.xz"
+
+debian_src_prepare() {
+	local debian_dir="${WORKDIR}"/debian
+	local debian_patches_dir="${debian_dir}"/patches
+
+	# Always parse the series file because not all Debian patches are
+	# (exclusively) numbered.
+	if [[ -f "${debian_patches_dir}"/series ]] ; then
+		readarray -t debian_patch_list < "${debian_patches_dir}"/series
+
+		mv "${debian_patches_dir}" "${debian_patches_dir}.orig" || die
+		mkdir "${T}"/debian_eclass_sorted || die
+
+		# We want to get the nice eapply output when it's given a directory,
+		# as it clearly delinates what the external patches are, so
+		# we rename the patches into an order eapply will Just Work with.
+		local patch_index
+		for ((patch_index=0; patch_index < ${#debian_patch_list[@]} ; patch_index++)); do
+			local patch=${debian_patch_list[${patch_index}]}
+			local new_patch_num=$(printf "%04d" ${patch_index})
```

Comment:
```suggestion
		local patch patch_index=0
		for patch in "${debian_patch_list[@]}"; do
			local new_patch_num=$(printf "%04d" $(( patch_index++ )))
```

(alternatively, you could replace the `(( ... ))` with `${!debian_patches[@]}` but I think the above is simpler)